client/actions.js

```diff
@@ -0,0 +1,308 @@
+const searchSubmitted = (search) => {
+  return { type: 'SEARCH_SUBMITTED', search }
+}
+
+const renderResults = (results) => {
+  return { type: 'RENDER_RESULTS', results }
+}
+
+const searchCleared = () => {
+  return { type: 'SEARCH_CLEARED' }
+}
+
+const search = searchString => {
+  const searchArray = searchString.trim().split(' ');
+  if (searchArray.length === 0) return;
+
+  return dispatch => {
+    dispatch(searchSubmitted(searchString));
+    if (searchArray.length === 1 || searchArray.length === 3) {
+      const resultsArray = [];
+      fetch(`/orgchartemployee/${searchArray[0]}`, {
+        headers: {'Content-Type': 'application/json'}
+      })
+        .then(response => response.json())
+        .then(response => {
+          if (response.id === searchArray[0]) {
+            resultsArray.push(response);
+          }
+        })
+        .then ( () => {
+          fetch(`/searchname/${searchArray[0]}`, {
+            headers: {'Content-Type': 'application/json'}
+          })
+            .then(response => response.json())
+            .then(response => {
+              if (response) {
+                response.forEach(result => {
+                  resultsArray.push(result);
+                })
+              }
+              dispatch(renderResults(resultsArray))
+              document.getElementById('search').reset();
+              dispatch(searchCleared());
+            })
+        })
+      }
+    if (searchArray === 2) {
+      fetch(`/searchnames/${searchArray[0]}/${searchArray[1]}`, {
+        headers: {'Content-Type': 'application/json'}
+      })
+        .then(response => response.json())
+        .then(response => {
+          dispatch(renderResults(response))
+        })
+    }
+
+  }
+}
+
+const employeeSubmitted = () => {
+  return { type: 'EMPLOYEE_SUBMITTED' }
+}
+
+const employeeSaved = (response) => {
+  return { type: 'EMPLOYEE_SAVED', response }
+}
+
+const employeeFailure = () => {
+  return { type: 'EMPLOYEE_FAILURE' }
+}
+
+const formCleared = () => {
+  return { type: 'FORM_CLEARED' }
+}
+
+const saveEmployee = employee => {
+  return dispatch => {
+    dispatch(employeeSubmitted());
+
+    fetch('/newemployee/', {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify(employee)
+    })
+      .then(response => response.json())
+      .then(response => {
+        if (response.error) {
+          alert(response.error);
+          dispatch(employeeFailure());
+        }
+        else if (response.id) {
+          alert(`Employee ${response.id} ${response.first_name} ${response.last_name} succcessfully created!`)
+          dispatch(employeeSaved(response));
+        }
+
+      })
+      .then( () => {
+        if (document.getElementById('employee')) {
+          document.getElementById('employee').reset();
```

Comment:
What is this `#employee` element? Any reason you can't update the state with an action instead of reading from the DOM?